components/ide-service/pkg/server/server.go

```diff
@@ -445,27 +460,23 @@ func (s *IDEServiceServer) ResolveWorkspaceConfig(ctx context.Context, req *api.
 			}
 		}
 
-		// we always need WebImage for when the user chooses a desktop ide
-		resp.WebImage = getUserIDEImage(defaultIde)
-		resp.IdeImageLayers = getUserImageLayers(defaultIde)
-
 		var desktopImageLayer string
 		var desktopUserImageLayers []string
 		if chosenIDE.Type == config.IDETypeDesktop {
-			desktopImageLayer = getUserIDEImage(chosenIDE)
-			desktopUserImageLayers = getUserImageLayers(chosenIDE)
+			desktopImageLayer = getUserIDEImage(resultingIdeName)
+			desktopUserImageLayers = getUserImageLayers(resultingIdeName)
 		} else {
-			resp.WebImage = getUserIDEImage(chosenIDE)
-			resp.IdeImageLayers = getUserImageLayers(chosenIDE)
+			resp.WebImage = getUserIDEImage(resultingIdeName)
+			resp.IdeImageLayers = getUserImageLayers(resultingIdeName)
 		}
 
 		// TODO (se) this should be handled on the surface (i.e. server or even dashboard) and not passed as a special workspace context down here.
-		ideName, referrer := s.resolveReferrerIDE(ideConfig, wsContext, userIdeName)
+		ideName, _ := s.resolveReferrerIDE(ideConfig, wsContext, userIdeName)
```

Comment:
(off-topic) i believe it should be deleted, we don't oass refererrer context anyore, but resovle it on dashboard